google-cloud-storage/src/main/java/com/google/cloud/storage/GrpcStorageImpl.java

```diff
@@ -682,9 +750,78 @@ private Blob internalCreate(
 
   /** Bind some decoders for our "Syntax" classes to this instance of GrpcStorageImpl */
   private final class SyntaxDecoders {
+
     final Decoder<Object, Blob> blob =
         o -> codecs.blobInfo().decode(o).asBlob(GrpcStorageImpl.this);
     final Decoder<com.google.storage.v2.Bucket, Bucket> bucket =
         b -> codecs.bucketInfo().decode(b).asBucket(GrpcStorageImpl.this);
   }
+
+  private static class TransformPageDecorator<
+          RequestT,
+          ResponseT,
+          ResourceT,
+          PageT extends AbstractPage<RequestT, ResponseT, ResourceT, PageT>,
+          ModelT>
+      implements Page<ModelT> {
+
+    private final PageT page;
+    private final Function<ResourceT, ModelT> translator;
```

Comment:
If we change this field type from `Function<ResourceT, ModelT>` to `Decoder<ResourceT, ModelT>` then we can simplify each of the list methods a bit by directly pointing to the decoders, including the new `syntaxDecoders` that were added during my retry refactor last week.

```
    return new TransformPageDecorator<>(page, syntaxDecoders.bucket);
    return new TransformPageDecorator<>(page, syntaxDecoders.blob);
    return new TransformPageDecorator<>(page, codecs.hmacKeyMetadata());
```